server/src/main/scala/org/elasticmq/server/config/ElasticMQServerConfig.scala

```diff
@@ -68,47 +71,75 @@ class ElasticMQServerConfig(config: Config) extends Logging {
 
   val restStatisticsConfiguration = new RestStatisticsConfiguration
 
+  private val queuesStorage = config.getConfig("queues-storage")
+  val queuesStoragePath: String = queuesStorage.getString("path")
+  val queuesStorageEnabled: Boolean = queuesStorage.getBoolean("enabled")
+
+  def readQueuesToLoad(persistedQueues: Seq[CreateQueue]): Seq[CreateQueue] = {
+    val persistedQueuesName = persistedQueues.map(_.name)
+    val baseQueues = createBaseQueues.filterNot(queue => persistedQueuesName.contains(queue.name))
+    persistedQueues ++ baseQueues
+  }
+
+  val createBaseQueues: Seq[CreateQueue] = {
+    val baseQueues: mutable.Map[String, ConfigValue] = config
+      .getObject("queues")
+      .asScala
+    createQueuesFromConfig(baseQueues)
+  }
+
+  private val persistedQueuesConfig: Option[Config] =
+    if (queuesStorageEnabled)
+      Some(
+        ConfigFactory
+          .parseFile(new File(queuesStoragePath))
+      )
+    else None
+
+  def createPersistedQueues(backupFile: Option[Config] = persistedQueuesConfig): List[CreateQueue] =
+    backupFile match {
+      case Some(file) =>
+        val queuesConfig: mutable.Map[String, ConfigValue] = file
+          .getObject("queues")
+          .asScala
+        createQueuesFromConfig(queuesConfig)
+      case None => Nil
+    }
 
-  val createQueues: List[CreateQueue] = {
+  def createQueuesFromConfig(queuesConfig: mutable.Map[String, ConfigValue]): List[CreateQueue] = {
```

Comment:
if this doesn't depend on other internal state of `ElasticMQServerConfig`, maybe let's move it to the companion object so that we're sure it's a "stateless" function?